.github/workflows/run-tests.yml

```
@@ -116,12 +116,14 @@ jobs:
         run: git fetch origin ${{ github.event.pull_request.base.ref }}:${{ github.event.pull_request.base.ref }}
 
       - name: Compute merge base
+        if: github.ref != 'refs/heads/main'
```

how about
```suggestion
        if: ${{ github.event_name == 'pull_request' }}
```
?
https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#example-usage-of-the-github-context